Pipfile

```diff
@@ -37,7 +37,6 @@ sphinx = ">=1.8,<2"
 [packages]  # Make sure to keep in sync with setup.py requirements.
 arrow = ">=0.15.0,<1"
 funcy = ">=1.7.3,<2"
-dataclasses = ">=0.6"
```

Comment:
A con here is that if someone was using python3.6 to develop locally then they'll get import error. 

Technically, the Pipfile requires python 3.8, but I think that pipenv still works if you use an incorrect version of python. Also even though our `CONTRIBUTING.rst` information states that developers should use python3.8 for local development, our `TROUBLESHOOTING.rst` has documentation that makes it sound that you can use different python versions:
"# Please modify command to match the version of python you use for development."
"sudo apt-get install python3.6-dev"

If we are going to proceed with this PR, do you mind updating the `TROUBLESHOOTING.rst` and the readthedocs equivalent to make it clearer that developers are only supposed to use python3.8 for local development?

Reply:
Updated all docs as suggested.

Reply:
Thanks!